File: WebApplication2/WebApplication2/Security/FakeUserMapper.vb
Imports Nancy.Authentication.Forms
Imports WebApplication2.Services

Namespace Security
    Public Class FakeUserMapper
        Implements IUserMapper

        Private ReadOnly _userService As UserService

        Public Sub New(userService As UserService)
            _userService = userService
        End Sub


        Public Function GetUserFromIdentifier(ByVal identifier As Guid, ByVal context As Nancy.NancyContext) As Nancy.Security.IUserIdentity Implements IUserMapper.GetUserFromIdentifier
            Dim user = _userService.GetById(identifier)
            Return New AuthenticatedUser() With
            {
                .UserName = user.Name,
                .Claims = user.Claims
            }
        End Function

    End Class
End Namespace

File: NancyDemo.Csharp/Security/FakeUserMapper.cs
using System;
using Nancy.Authentication.Forms;
using NancyDemo.Csharp.Services;

namespace NancyDemo.Csharp.Security
{
    public class FakeUserMapper :IUserMapper
    {
        private readonly UserService _userService;

        public FakeUserMapper(UserService userService)
        {
            _userService = userService;
        }


        public Nancy.Security.IUserIdentity GetUserFromIdentifier(Guid identifier, Nancy.NancyContext context)
        {
            var user = _userService.GetById(identifier);
            return new AuthenticatedUser
                {
                    UserName = user.Name,
                    Claims = user.Claims,
                    RealName = user.RealName
                };
        }

    }
}
